notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/InstallationPutRequest.java

```diff
@@ -0,0 +1,185 @@
+package com.microsoft.windowsazure.messaging.notificationhubs;
+
+import android.os.Build;
+import android.util.Base64;
+
+import com.android.volley.NetworkResponse;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.HttpHeaderParser;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.microsoft.windowsazure.messaging.BuildConfig;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.URLEncoder;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * Custom Android Volley Request to UPSERT an Installation records with the Azure Notification Hub backend.
+ *
+ * It was built specifically to work with API Version 2020-06.
+ */
+class InstallationPutRequest extends JsonObjectRequest {
+    private final static String API_VERSION = "2020-06";
+    private final static String USER_AGENT = String.format("NOTIFICATIONHUBS/%s (api-origin=AndroidSdkV1FcmV%s; os=%s; os_version=%s;)",
+            API_VERSION, BuildConfig.VERSION_NAME, "Android", Build.VERSION.RELEASE);

```

Comment:
This is not quite right, because it's not always Android, it may be FCM, GCM, etc, see previous code here https://github.com/Azure/azure-notificationhubs-android/blob/master/notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/PnsSpecificRegistrationFactory.java#L173-L192

Reply:
Chatting more about this offline. Basically, the short of it is that at the moment we're only supporting Firebase. For that reason, this is out of scope for this PR. We're meeting today to make sure we are all in agreement about the cadence at which we add Baidu/ADM support to this newer namespace.

Reply:
At our meeting today, everyone present agreed that we should support both ADM and Baidu, but that we shouldn't wait to release 1.0.0 until we do. In the coming months we'll release follow up versions with the new platforms.

If you're reading this, really need ADM/Baidu support and want to use the new Installation stuff, please [let us know](https://github.com/Azure/azure-notificationhubs-android/issues/new).